application/Application.d.ts

```diff
@@ -0,0 +1,128 @@
+// There is no Interger in Typescript so it has been classed as a number but the Interger is still expected. 
+type Interger = number;
+
+declare class Application {
+  hostUrl: any;
+  apiKey: any;
+
+  constructor(host: URL, key: String);
+  processError(err: { status: undefined; statusCode: undefined }):
+    | number
+    | {
+        status: undefined;
+        statusCode: undefined;
+      }
+    | undefined;
+
+  getErrorCode(err: {
+    status?: undefined;
+    statusCode?: undefined;
+    message?: any;
+    charAt?: any;
+    length?: any;
+  }): number | undefined;
```

Comment:
Unknown so set as any or undefined.

Reply:
Same again, getErrorCode not intended for use by the user